test/e2e/self_hosted.go

```diff
@@ -184,6 +196,27 @@ func SelfHostedSpec(ctx context.Context, inputGetter func() SelfHostedSpecInput)
 		})
 		Expect(controlPlane).ToNot(BeNil())
 
+		// After the move check that there were no unexpected rollouts.
+		Consistently(func() bool {
+			postMoveMachineList := &clusterv1.MachineList{}
+			err = selfHostedClusterProxy.GetClient().List(
+				ctx,
+				postMoveMachineList,
+				client.InNamespace(namespace.Name),
+				client.MatchingLabels{clusterv1.ClusterLabelName: workloadClusterName},
+			)
+			Expect(err).NotTo(HaveOccurred())
+			oldObjects := []client.Object{}
+			for _, machine := range preMoveMachineList.Items {
+				oldObjects = append(oldObjects, &machine)
```

Comment:
this is unrecommended because of things like https://go.dev/play/p/8u0R9_rA-zJ

Reply:
Good point. Switching to 
```golang
oldObjects = append(oldObjects, &preMoveMachineList.items[i])
``` 
would fix the problem.